citizenvoice/apiapp/views.py

```diff
@@ -6,150 +6,229 @@
-            return queryset
-
-    # Get a Questions based on their survey ID
     @staticmethod
-    def GetQuestionsFromSurvey(survey_id):
-        queryset = SurveyViewSet.GetSurvey(survey_id)[0].question_set.all()
-        serializer_class = QuestionSerializer
+    def GetQuestionByID(id):
+        """
+        Get a specific Question based on its ID.
+
+        Parameters:
+            id (int): Question ID to be used for finding this Question.
+
+        Return: 
+            queryset: containing the Question instance with this id
+        """
+
+        queryset = Question.objects.filter(id=id)
         return queryset
 
-    # Get a specific Question based on its survey ID and order in the survey
     @staticmethod
-    def GetOrderedQuestionFromSurvey(survey_id, question_order):
-        queryset = SurveyViewSet.GetSurvey(survey_id)[0].question_set.all().filter(order=question_order)
-        serializer_class = QuestionSerializer
+    def GetQuestionBySurvey(survey_id): 
+        """
+        Get specific Questions based on its survey_id.
+
+        Parameters:
+            survey_id (int): Survey ID to be used for finding related Questions.
+
+        Return: 
+            queryset: containing the Question instance related to this survey
+        """
+
+        queryset = Question.objects.filter(survey=survey_id)
+        return queryset  
+
+    @staticmethod
+    def GetOrderedQuestionBySurvey(survey_id, question_order):
+        """
+        Get specific Questions based on its survey_id, and a specific order.
+
+        Parameters:
+            survey_id (int): Survey ID to be used for finding related Questions.
+            question_order (int): The order in which the questions in the Survey are to be displayed.
+
+        Return: 
+            queryset: containing the Question instance related to this survey, of a given order
+        """
+        queryset = Question.objects.filter(survey=survey_id, order=question_order)
         return queryset
+        
 
-# Create a ViewSet that queries all the instances of Survey in the database, and parse them through the serializer
 class SurveyViewSet(viewsets.ModelViewSet):
     """
-    Survey ViewSet used internally to query data from database. The following functions are defined in this viewset:
+    Survey ViewSet used internally to query data from database.
 
-    GetSurveys() - returns a set of all Survey instances in the database
+    """
 
-    GetSurvey(int id) - returns an instance of Survey with the ID that was provided in the function.
+    serializer_class = SurveySerializer
 
-    """
+    def get_queryset(response):
+        """
+        Returns a set of all Survey instances in the database.
+
+        Return:
+            queryset: containing all Survey instances
+        """
 
-    # Get all surveys
-    @staticmethod
-    def GetSurveys():
         queryset = Survey.objects.all().order_by('name')
-        serializer_class = SurveySerializer
         return queryset
```

Comment:
This will retrieve all surveys, regardless of the user (which also has to be there as earlier mentioned for the `Answer` model.
The `GetSurveyByAuthor` method (to be renamed now to `GetSurveyByDesigner`) is responsible for returning surveys associated to a specific user.